tokenizer/udic_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2015 ikawaha
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// 	You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tokenizer
+
+import (
+	"encoding/json"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+var testFile = "../_sample/userdic.txt"
+
+func TestNewUserDic01(t *testing.T) {
+	if _, e := NewUserDic(""); e == nil {
+		t.Error("expected error, but no occured\n")
+	}
+}
+
+func TestNewUserDicIndex01(t *testing.T) {
+	udic, e := NewUserDic(testFile)
+	if e != nil {
+		t.Fatalf("unexpected error: %v\n", e)
+	}
+	type tuple struct {
+		inp string
+		id  int
+		ok  bool
+	}
+	callAndRespose := []tuple{
+		{inp: "日本経済新聞", id: 0, ok: true},
+		{inp: "朝青龍", id: 1, ok: true},
+		{inp: "関西国際空港", id: 2, ok: true},
+		{inp: "成田国際空港", id: 9, ok: false},
+	}
+	for _, cr := range callAndRespose {
+		if ids := udic.dic.Index.Search(cr.inp); (len(ids) != 0) != cr.ok {
+			t.Errorf("got %v, expected %v\n", ids, cr.ok)
+		}
+	}
+}
+
+func TestNewUserDicRecords01(t *testing.T) {
+	r := UserDicRecords{
+		{
+			Text:   "日本経済新聞",
+			Tokens: []string{"日本", "経済", "新聞"},
+			Yomi:   []string{"ニホン", "ケイザイ", "シンブン"},
+			Pos:    "カスタム名詞",
+		},
+		{
+			Text:   "朝青龍",
+			Tokens: []string{"朝青龍"},
+			Yomi:   []string{"アサショウリュウ"},
+			Pos:    "カスタム人名",
+		},
+	}
+	udic, err := r.NewUserDic()
+	if err != nil {
+		t.Fatalf("user dic build error, %v", err)
+	}
+	if ids := udic.dic.Index.Search("日本経済新聞"); len(ids) != 1 {
+		t.Errorf("user dic search failed")
+	} else if !reflect.DeepEqual(udic.dic.Contents[ids[0]].Tokens, []string{"日本", "経済", "新聞"}) {
+		t.Errorf("got %+v, expected %+v", udic.dic.Contents[ids[0]].Tokens, []string{"日本", "経済", "新聞"})
+	}
+	if ids := udic.dic.Index.Search("関西国際空港"); len(ids) != 0 {
+		t.Errorf("user dic build failed")
+	}
+	if ids := udic.dic.Index.Search("朝青龍"); len(ids) == 0 {
```

Comment:
__[gometalinter]__
* [dupl] duplicate of udic_test.go:112-116

<!-- Comment by [SideCI](https://sideci.com) -->